### PR TITLE
feat(pipelines): add --pipeline-dir argument to cli

### PR DESCRIPTION
**Pull request by jdrouet**
Signed-off-by: Jérémie Drouet <jeremie.drouet@datadoghq.com>

### CODE REVIEW

File: src/cli.rs [hunk 1]
```diff
@@ -117,6 +117,19 @@ pub struct RootOpts {
     )]
     pub config_paths_yaml: Vec<PathBuf>,
 
+    /// Read pipeline configuration from files in one or more directories.
+    /// File format is detected from the file name.
+    ///
+    /// Files not ending in .toml, .json, .yaml, or .yml will be ignored.
```
Comment (spencergilbert) [inline src/cli.rs:123]: Did we determine this was the actual behavior?

Comment (jdrouet) [reply src/cli.rs:123]: considering it has to behave like the config files, yes.

File: src/config/loading.rs [hunk 2]
```diff
@@ -97,8 +97,16 @@ pub fn process_paths(config_paths: &[ConfigPath]) -> Option<Vec<ConfigPath>> {
...
+pub fn load_from_paths(
+    config_paths: &[ConfigPath],
+    pipeline_paths: &[PathBuf],
+) -> Result<Config, Vec<String>> {
+    let pipelines = if pipeline_paths.is_empty() {
+        let pipeline_paths = pipeline_paths_from_config_paths(config_paths);
+        load_pipelines_from_paths(&pipeline_paths)?
+    } else {
+        load_pipelines_from_paths(pipeline_paths)?
+    };
```
Comment (lukesteensen) [inline src/config/loading.rs:109]: Would it be worth extracting this pattern to its own function? Seems like we're doing this a lot. Maybe something like `load_pipelines_from_paths(pipeline_paths, config_paths)`?

**Review by lukesteensen — APPROVED** [review]
One suggestion, but overall this implementation seems reasonable to me.

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**